Firmware/ultralcd.cpp

```diff
@@ -7542,14 +7546,19 @@ static bool lcd_selftest_IRsensor(bool bStandalone)
 
 static void lcd_detect_IRsensor(){
     bool bAction;
-
+	bool loaded;
     bMenuFSDetect = true;                               // inhibits some code inside "manage_inactivity()"
-    bAction = lcd_show_fullscreen_message_yes_no_and_wait_P(_i("Is filament loaded?"), false, false);
-    if(bAction){
-        lcd_show_fullscreen_message_and_wait_P(_i("Please unload the filament first, then repeat this action."));////c=20 r=4
+	//! Check if filament is loaded. If it is loaded stop detection.
+	//! @todo Add autodetection with MMU2s
+	loaded = (digitalRead(IR_SENSOR_PIN) == 0);
```

Comment:
are you sure this `loaded` will work even if the fsensor is unconnected, damaged etc.?

Reply:
I get what you mean as I don't read the raw voltage value but just the digital value.
- Unconnected should result as digital 1 
- A broken one we don't know for sure what it would be.
But if you like we can keep the Fsensor detection asking for the status.